### ISSUE TITLE
target_container option not working as expected

### ISSUE THREAD

**Issue by coingraham** [aws/copilot-cli#4041]
Website Manifest
```
name: website
type: Load Balanced Web Service

http:
  path: '/'
  target_container: proxy

image:
  build: copilot/website/Dockerfile
  port: 80

cpu: 256
memory: 512
platform: linux/x86_64
count: 1

sidecars:
  proxy:
    port: 8443
    image: 123456789012.dkr.ecr.us-east-2.amazonaws.com/application-name/proxy
```

When I use the above manifest to setup an sidecar proxy, the Target Group gets setup with the original container port instead of the target container port.  Here's a look at the section of the CFT that matters:
```
      LoadBalancers:
        - ContainerName: !Ref TargetContainer <--- Here the target container is used
          ContainerPort: !Ref TargetPort <--- Here the target port is used
          TargetGroupArn: !Ref TargetGroup
      ServiceRegistries:
        - RegistryArn: !GetAtt DiscoveryService.Arn
          Port: !Ref ContainerPort <--- The original container port.  Should probably be the target port.
  TargetGroup:
    Metadata:
      'aws:copilot:description': "A target group to connect the load balancer to your service"
    Type: AWS::ElasticLoadBalancingV2::TargetGroup
    Properties:
      HealthCheckPath: / # Default is '/'.
      Port: !Ref ContainerPort <--- This should be the target port
      Protocol: HTTP
```

**Comment by dannyrandall** [comment]
Hey @coingraham! Thanks so much for reporting this. I'm able to reproduce the bug - you're right, it should be `TargetPort` in those locations.

While you're waiting for a fix, is it possible to have the image/sidecar run on the same port? If so, the traffic will go to the proxy.

**Comment by dannyrandall** [comment]
Hey @coingraham - thanks so much for reporting this! I was able to reproduce the bug. You're absolutely right that those fields should be `TargetPort`.

Sorry about this!🙇 We'll get to work on a fix!

**Comment by dannyrandall** [comment]
Hey @coingraham, I've done some more testing and just wanted to double check that you're not actually running into any issues reaching your services, right?

**Comment by coingraham** [comment]
Yes it is failing.  So long story short is that I simplified the above example to showcase the bug easily and clearly.  The fact is that I'm using the new HTTPS functionality that was released a few days ago.  It sets the Healthcheck protocol based on the **target** port.  So when you combine the target sidecar container on 443 and the original container on 80, you end up with a healthcheck on 80/https which doesn't succeed for either container.  So if you take the above example and make the sidecar port 443, you'll see it fail completely.

**Comment by dannyrandall** [comment]
Ohhh - so you're saying that if you configure:
- Primary container on port 80
- Sidecar on port 443
- `target_container` = the sidecar container
- `http.healthcheck.port` to 80

The problem then is that even though you've changed the health check port to 80, it's still setting the health check protocol to https? So the health checks fail going to the primary container?

**Comment by coingraham** [comment]
Currently it will hold the original container port of 80 for both the target group and healthcheck but uses the target port of 443 to determine the protocol of HTTPS.  So connections and healthchecks fail.

I might be able to hard code the healthcheck to 443 but that would put me in the worst case situation where the ALB thinks the target group is working but it actually isn't.  Here's a look at my non-functioning manifest:

```
name: website
type: Load Balanced Web Service

http:
  alias: this.example.com
  path: '/'
  target_container: proxy

image:
  build: copilot/website/Dockerfile
  port: 80

cpu: 256
memory: 512
platform: linux/x86_64
count: 1

sidecars:
  proxy:
    port: 443
    image: 123456789012.dkr.ecr.us-east-2.amazonaws.com/bugfix/proxy
```

**Comment by paragbhingre** [comment]
The fix is now released in v1.22.1: https://github.com/aws/copilot-cli/releases/tag/v1.22.1 :tada:
